nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,281 @@
+import string
+import os
+import pprint
+
+import pandas as pd
+from nilearn.datasets import load_mni152_template
+from nilearn.plotting import plot_stat_map
+
+from nistats.reporting import (
+    plot_design_matrix,
+    get_clusters_table,
+    )
+
+
+html_template_path = os.path.join(os.path.dirname(__file__), 'report_template.html')
+
+
+def make_glm_report(output_path,
+                    model,
+                    contrasts,
+                    threshold=3.09,
+                    bg_img='MNI 152 Template',
+                    display_mode='z'):
+    """ Creates an HTML page which shows all important aspects
+    of a fitted GLM.
+    
+    Parameters
+    ----------
+    output_path: String (Path)
+        The file path to which the HTML report will be saved.
+    
+    model: FirstLevelModel or SecondLevelModel object
+        A fitted first or second level model object.
+        
+    contrasts: Dict[string, ndarray]
+    
+    threshold: float
+        Default is 3.09
+        
+    bg_img: Niimg
+        Default is the MNI152 template
+        
+    display_mode: String
+        Default is 'z'.
+        
+    Returns
+    -------
+    None
```

Comment:
Then it will be doing more than reporting. As I envisioned it, the reporter shouldn't do much more than reporting. 
If the users want the contrasts they can do that outside.  We are computing the contrasts, as a convenience to the user (and even that feels uncomfortable to me), but the sheer cleanliness of the script and convenience can be considered more important there. 
I don't feel super comfortable with one function doing so many things (reporting, computing contrasts, saving HTML reports, returning contrasts). 
It doesn't track with all the stuff I have seen and read on good design.

Reply:
I think that simplifies greatly the API to make the function compute contrast. In most cases, this is a rather inexpensive operation ---but there are exceptions.

I'm not sure that they should be returned. I tend to see reports as an endpoint.